pkgs/by-name/pr/protonmail-bridge-gui/package.nix

```diff
@@ -0,0 +1,132 @@
+    "-DBRIDGE_APP_FULL_NAME='Proton Mail Bridge'"
+    "-DBRIDGE_VENDOR='Proton AG'"
+    "-DBRIDGE_REVISION=${src.rev}"
+    "-DBRIDGE_TAG=v${version}"
+    "-DBRIDGE_BUILD_ENV=dev"
+    "-DBRIDGE_APP_VERSION=${version}"
+  ];
+
+in
+stdenv.mkDerivation {
+  inherit pname version src patches cmakeFlags;
+
+  nativeBuildInputs = [
+    pkg-config
+    cmake
+    ninja
+    qt6.qtbase
+    qt6.qtdeclarative
+    qt6.qtwayland
+    qt6.qtsvg
+    qt6.wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    libsecret
+    grpc
+    protobuf
+    zlib
+    gtest
+    sentry-native
+  ];
+
+  postPatch = ''
+    # bypass `vcpkg` by deleting lines that `include` internal/frontend/bridge-gui/BridgeSetup.cmake
+    find internal/frontend/bridge-gui -type f -name "CMakeLists.txt" -exec sed -i "/BridgeSetup\\.cmake/d" {} \;
+
+    # modify DeployLinux.cmake to use the available ICU version
+    sed -i 's/libicu\(i18n\|uc\|data\)\.so\.56/libicu\1.so/g' internal/frontend/bridge-gui/bridge-gui/DeployLinux.cmake
+  '';
+
+  dontUseCmakeConfigure = true; # disable the automatic configure phase
+
+  configurePhase = ''
+    runHook preConfigure
+
+    export QT_PLUGIN_PATH="${qt6.qtbase}/${qt6.qtbase.qtPluginPrefix}"
+
+    pushd internal/frontend/bridge-gui
+    cmake \
+      ${lib.concatStringsSep " " cmakeFlags} \
+      -G Ninja \
+      -S . \
+      -B build
+    popd
+
+    runHook postConfigure
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    pushd internal/frontend/bridge-gui
+    cmake --build build --verbose
+    popd
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    # Install the GUI binary
+    install -Dm755 internal/frontend/bridge-gui/build/bridge-gui/bridge-gui $out/lib/bridge-gui
+
+    # Symlink the backend binary from the protonmail-bridge (CLI) package
+    ln -s ${protonmail-bridge}/bin/protonmail-bridge $out/lib/bridge
+
+    # Symlink the GUI binary
+    mkdir -p $out/bin
+    ln -s $out/lib/bridge-gui $out/bin/protonmail-bridge
+
+    # Install assets
+    install -Dm644 dist/proton-bridge.desktop -t $out/share/applications
+    install -Dm644 dist/bridge.svg $out/share/icons/hicolor/scalable/apps/protonmail-bridge.svg
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/ProtonMail/proton-bridge";
+    changelog = "https://github.com/ProtonMail/proton-bridge/blob/${src.rev}/Changelog.md";
+    downloadPage = "https://github.com/ProtonMail/proton-bridge/releases";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ daniel-fahey ];
+    description = "Qt-based GUI to use your ProtonMail account with your local e-mail client";
+    longDescription = ''
+      An application that runs on your computer in the background and seamlessly encrypts
+      and decrypts your mail as it enters and leaves your computer.
+
+      To work, use secret-service freedesktop.org API (e.g. Gnome keyring) or pass.
+    '';
+    mainProgram = "protonmail-bridge";
```

Comment:
`meta.mainProgram` is just a string naming the executable in `$out/bin`

Reply:
I meant changing the executable and this also represent the name of the "main exec
utable" IIUC so if it changes this should also change. 

Not sure if it should be a big concern or not though, for instance the name clash would be a problem if installing both packages via system packages.

Reply:
Hi @MrFreezeex thanks for reviewing my PR! I deliberately made `protonmail-bridge` and `protonmail-bridge-gui` collide (by having the same binary name). The binary provided by `protonmail-bridge-gui` can still be used via its CLI i.e. `protonmail-bridge --cli`, and I wanted it to be an optional drop-in replacement, so that anyone who has used `protonmail-bridge` in their scripts / services can still use the binary provided by `protonmail-bridge-gui`. I've updated the `meta.description` and `meta.longDescription` to make this clearer. My reasoning is that the CLI only package will be more resilient to version updates, and my +GUI package will likely need new patches etc., so safer to keep them separate, and easier to maintain.

Reply:
Ah ok I see very clear thank you for the explanation 🙏

Reply:
Thanks for the contribution, looking forward to this - gave it a try myself some time ago but gave up fighting with vcpkg ;)

Do you plan to extend the derivations then to make the gui-derivation reuse the cli-one as base and only extend the needed attributes to have one variant/derivation to build the cli from so if I use the other package I'm sure the cli still works?

or use as withGui attribute like we do https://github.com/NixOS/nixpkgs/blob/00fabffe1cb59794afe6ccdff9b6076f7c98b089/pkgs/by-name/ls/lshw/package.nix#L9 and do two callPackages one with and one without?

Reply:
> The binary provided by `protonmail-bridge-gui` can still be used via its CLI i.e. `protonmail-bridge --cli`

@MrFreezeex I was actually mistaken here, it works in other distros, but they use the launcher (with the auto-update stuff) to do this, so I've gone with your suggestion of using a separate `protonmail-bridge-gui` binary, and have patched the desktop entry to use it, and also removed the non-functional desktop entry from the CLI derivation.